src/linear_algebra/finsupp.lean

```diff
@@ -822,6 +822,39 @@ end prod
 
 end finsupp
 
+section fintype
+
+variables (α M R : Type*) [semiring R] [add_comm_monoid M] [module R M] (v : α → M) [fintype α]
+
+/-- A variant of `finsupp.total -/
+noncomputable
+def fintype.total : (α → R) →ₗ[R] M :=
+(finsupp.total α M R v).comp (finsupp.linear_equiv_fun_on_fintype R R α).symm.to_linear_map
```

Comment:
I think it would make sense to define this directly rather than going via finsupp.total; it would be computable and likely definitionally more convenient